lib/stdlib/src/supervisor.erl

```diff
@@ -224,11 +225,13 @@ delete_child(Supervisor, Id) ->
 %%          way (maybe killed).
 %%-----------------------------------------------------------------
 
+-type terminatechild_err() :: 'not_found' | 'simple_one_for_one'.
+-type terminatechild_ret() :: 'ok' | {'error', terminatechild_err()}.
+
 -spec terminate_child(SupRef, Id) -> Result when
       SupRef :: sup_ref(),
       Id :: pid() | child_id(),
-      Result :: 'ok' | {'error', Error},
-      Error :: 'not_found' | 'simple_one_for_one'.
+      Result :: terminatechild_ret().
```

Comment:
Why do we need the subtype terminatechid_err ?

Reply:
i will revert it